locator/token_metadata.cc

```diff
@@ -929,12 +929,36 @@ future<> topology_change_info<NodeId>::clear_gently() {
 
 template <typename NodeId>
 generic_token_metadata<NodeId>::generic_token_metadata(std::unique_ptr<token_metadata_impl<NodeId>> impl)
-    : _impl(std::move(impl)) {
```

Comment:
> token_metadata: support host_id-based version
> In this commit we enhance token_metadata with a pointer to its new version.
> The idea is that in the following commits we'll go over all token_metadata
> modifications and make the corresponding modifications to the new version

"new version" here is ambiguous; here you refer to the new type, not a new version of a token_metadata struct populated with different values.